app/frontend/components/pages/delegations/delegatePage.js

```diff
@@ -0,0 +1,159 @@
+const {h} = require('preact')
+const connect = require('unistore/preact').connect
+const actions = require('../../../actions')
+const tooltip = require('../../common/tooltip')
+const printAda = require('../../../helpers/printAda')
+const {AdaIcon} = require('../../common/svg')
+
+const formatStakePoolInfo = (className, message) => {
+  return h('div', {class: `stake-pool-info ${className}`}, message)
+}
+
+const CalculatingFee = () => h('div', {class: 'validation-message send'}, 'Calculating fee...')
+
+const getStakePoolValidationMessage = (stakePool, stakePools) => {
+  let className, message
+
+  const addressUsages = stakePools.filter((pool) => pool.address === stakePool.address)
+  const isDuplicate = addressUsages.length > 1 && stakePool.ID !== addressUsages[0].ID
+
+  if (stakePool.valid) {
+    if (isDuplicate) {
+      className = 'invalid'
+      message = 'Duplicate stake pool'
+    } else if (stakePool.percent === 0) {
+      className = 'valid warning'
+      message = `${stakePool.name} | Delegate a non-zero amount`
+    } else {
+      className = 'valid'
+      message = `${stakePool.name}`
+    }
+  } else {
+    className = 'invalid'
+    message = stakePool.address === '' ? '' : 'Invalid address'
+  }
+  return [className, message]
```

Comment:
Why don't we return object instead of an array? That way it would simplify code a lot

Reply:
Changed.